### ISSUE TITLE
PreferredSource prevents route from being created by systemd-networkd

### ISSUE THREAD

**Issue by SF2311** [systemd/systemd#31950]
### systemd version the issue has been seen with

255.4-1+b1 and 252.22-1~deb12u1

### Used distribution

Debian (sid and bookworm)

### Linux kernel version used

6.7.9-amd64 / 6.1.0-18-amd64

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

A default route being created as specified in the .network file.

### Unexpected behaviour you saw

When `PreferredSource` is set, the route is not created.

### Steps to reproduce the problem

I have a simple VPN Setup, with a physical interface which gets its IP address over DHCP:
```
[Match]
Name=enp1s0

[Network]
DHCP=ipv4
IPForward=True
KeepConfiguration=dhcp

[DHCPv4]
RouteTable=2
```
Additionally, I have a WireGuard interface, that is configured like this:
```
[Match]
Name=wg_test

[Address]
Address=10.53.16.8/32
Peer=192.168.30.1/32

[Link]
MTUBytes=1500

[Route]
Destination=0.0.0.0/0
Gateway=192.168.30.1
PreferredSource=10.53.16.8

[RoutingPolicyRule]
Priority=1
IncomingInterface=lo
IPProtocol=udp
DestinationPort=51801
Table=2
```

This will lead to the following situation in the main routing table:
```
$  ip r s
192.168.30.1 dev wg_test proto kernel scope link src 10.53.16.8
```

However, if I remove the `PreferredSource` directive, I get the expected behavior:
```
$ cat wg_test.network
[Match]
Name=wg_test

[Address]
Address=10.53.16.8/32
Peer=192.168.30.1/32

[Link]
MTUBytes=1500

[Route]
Destination=0.0.0.0/0
Gateway=192.168.30.1
#PreferredSource=10.53.16.8

[RoutingPolicyRule]
Priority=1
IncomingInterface=lo
IPProtocol=udp
DestinationPort=51801
Table=2

$ systemctl restart systemd-networkd
$ ip r s
default via 192.168.30.1 dev wg_test proto static
192.168.30.1 dev wg_test proto kernel scope link src 10.53.16.8
```

On Debian 11 with `247.3-7+deb11u4` this setup works just fine.

[systemd-networkd.log](https://gist.github.com/SF2311/92344a80f6c90a2b81489f093dabe03d)

**Comment by Rookeur** [comment]
+1 I have the same issue

**Comment by yuwata** [comment]
Fix is waiting in #33875.
